drivers/pds/lib/utils.go

```diff
@@ -914,6 +914,30 @@ func CreatepostgresqlWorkload(dnsEndpoint string, pdsPassword string, scalefacto
 	return deployment, err
 }
 
+func CollectPodLogsandValidateWorkloads() error {
+	pods, err := k8sCore.GetPods("automation", nil)
+	if err != nil {
+		log.Errorf("An Error occured while getting pods")
+	}
+	failure_messages := []string{"Name or service not known", "could not translate host name", "Connection refused"}
+	for _, pod := range pods.Items {
+		if pod.Name == "pgload-slcww-64b4b7ffdd-bjt7w" {
+			podlogs, err := k8sCore.GetPodLog(pod.Name, pod.Namespace, &corev1.PodLogOptions{})
+			if err != nil {
+				log.Errorf("An Error occured while getting pod logs")
+			}
+			log.Infof(podlogs)
+			for _, messages := range failure_messages {
+				if strings.Contains(podlogs, messages) {
+					log.Infof("Unable to connect to the database")
```

Comment:
If we match the expected error message to the logs in pod, shouldn't we fail here? If we do fail here we don't need that `break` since we will just return if we find expected error